arena.cpp

```diff
@@ -56,41 +47,40 @@ void SnakeNode::copyNode(SnakeNode next)
 
 
 //These function returns a new node in different orientation
-SnakeNode SnakeNode::moveUp()
+SnakeNode SnakeNode::moveUp() const
 {
-    return SnakeNode(row-1, col);
+    return {row-1, col};
 }
 
-SnakeNode SnakeNode::moveDown()
+SnakeNode SnakeNode::moveDown() const
 {
-    return SnakeNode(row+1, col);
+    return {row+1, col};
 }
 
-SnakeNode SnakeNode::moveLeft()
+SnakeNode SnakeNode::moveLeft() const
 {
-    return SnakeNode(row, col-1);
+    return {row, col-1};
 }
 
-SnakeNode SnakeNode::moveRight()
+SnakeNode SnakeNode::moveRight() const
 {
-    return SnakeNode(row, col+1);
+    return {row, col+1};
 }
 
 
-
-Snake::Snake(int row, int col, int ori): orientation{ori}, INCREASE(0)
+Snake::Snake(const int row, const int col, const int ori): orientation{ori}, INCREASE(0)
 {
-    body.push_front(SnakeNode(row, col));
+    body.emplace_front(row, col);
```

Comment:
Implicit construction based on template parameter of "std::list<SnakeNode> body"